lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -166,17 +163,12 @@ def get_queryset(self):
         if course_ids:
             course_ids = course_ids.split(',')
             for course_id in course_ids:
-                course_key = CourseKey.from_string(course_id)
-                course_descriptor = courses.get_course(course_key)
-                results.append(course_descriptor)
+                results.append(CourseOverview.get_from_id(CourseKey.from_string(course_id)))
         else:
-            results = modulestore().get_courses()
-
-        # Ensure only course descriptors are returned.
-        results = (course for course in results if course.scope_ids.block_type == 'course')
+            results = CourseOverview.objects.all()
```

Comment:
Do we need all the data stored in this table, or just a certain set of fields from it?